google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -450,6 +469,13 @@ private TransactionContext internalBegin() {
     }
 
     private SpannerException handleSessionNotFound(SessionNotFoundException e) {
+      sessionReplacementCount++;
+      if (sessionReplacementCount > MAX_SESSION_REPLACES) {
+        throw SpannerExceptionFactory.newSpannerException(
+            e.getErrorCode(),
+            "Could not replace invalidated session because the maximum number of times the session may be replaced was exceeded",
+            e);
```

Comment:
same as above.. just leave this comment open until there is a better strategy defined for prev instance